src/ResourceManager/Sql/Commands.Sql/Security/Cmdlet/Auditing/SetAzureSqlDatabaseAuditingPolicy.cs

```diff
@@ -0,0 +1,122 @@
+        [ValidateSet(Constants.Primary, Constants.Secondary, IgnoreCase = false)]
+        [ValidateNotNullOrEmpty]
+        public string StorageKeyType { get; set; }
+
+        protected override bool WriteResult() { return PassThru; }
+
+        protected override DatabaseAuditingPolicyModel UpdateModel(DatabaseAuditingPolicyModel model)
+        {
+            base.UpdateModel(model);
+            model.AuditState = AuditStateType.Enabled;
+            model.UseServerDefault = UseServerDefaultOptions.Disabled;
+            if (StorageAccountName != null)
+            {
+                model.StorageAccountName = StorageAccountName;
+            }
+            if (!string.IsNullOrEmpty(StorageKeyType)) // the user enter a key type - we use it (and running over the previously defined key type)
+            {
+                model.StorageKeyType = (StorageKeyType == Constants.Primary) ? StorageKeyKind.Primary : StorageKeyKind.Secondary;
+            }
+
+            ProcessShortcuts();
+            if (EventType != null) // the user provided event types to audit, we use it
+            {
+                
+                Dictionary<string, AuditEventType> events = new Dictionary<string, AuditEventType>(){                
+                    {Constants.DataAccess, AuditEventType.DataAccess},
+                    {Constants.DataChanges, AuditEventType.DataChanges},
+                    {Constants.SecurityExceptions, AuditEventType.SecurityExceptions},
+                    {Constants.RevokePermissions, AuditEventType.RevokePermissions},
+                    {Constants.SchemaChanges, AuditEventType.SchemaChanges}
+                };
+                model.EventType = EventType.Select(s => events[s]).ToArray();
+            }
+            return model;
+        }
+
+        private void ProcessShortcuts()
+        {
+            if(EventType == null || EventType.Length == 0)
+            {
+                return;
+            }
+            if(EventType.Length == 1)
+            {
+                if(EventType[0] == Constants.None)
+                {
+                    EventType = new string[]{};
+                }
+                else if(EventType[0] == Constants.All)
+                {
+                    EventType = new string[]{Constants.DataAccess, Constants.DataChanges, Constants.SecurityExceptions, Constants.RevokePermissions, Constants.SchemaChanges};
+
+                }
+            }
+            else
+            {
+                if(EventType.Contains(Constants.All))
+                {         
+                  throw new Exception(string.Format(Resources.InvalidEventTypeSet, Constants.All));
+                }
+                if(EventType.Contains(Constants.None))
+                {
+                    throw new Exception(string.Format(Resources.InvalidEventTypeSet, Constants.None));
+                }
+            }
+        }
+    }
+
+
```

Comment:
please remove extra blank lines.

'}' should not be preceded by a blank line anywhere.